frontend/appflowy_tauri/src/appflowy_app/components/layout/NavigationPanel/NavItem.hooks.ts

```diff
@@ -0,0 +1,271 @@
+  };
+
+  const changePageTitle = async (newTitle: string) => {
+    await service.update({ name: newTitle });
+    appDispatch(pagesActions.renamePage({ id: page.id, newTitle }));
+  };
+
+  const closeRenamePopup = () => {
+    setShowRenamePopup(false);
+  };
+
+  const deletePage = async () => {
+    closePopup();
+    await service.delete();
+    appDispatch(pagesActions.deletePage({ id: page.id }));
+  };
+
+  const duplicatePage = async () => {
+    closePopup();
+    await service.duplicate();
+  };
+
+  const closePopup = () => {
+    setShowPageOptions(false);
+    setShowNewPageOptions(false);
+  };
+
+  const onPageClick = (page: IPage) => {
+    const pageTypeRoute = (() => {
+      switch (page.pageType) {
+        case ViewLayoutPB.Document:
+          return 'document';
+        case ViewLayoutPB.Grid:
+          return 'grid';
+        case ViewLayoutPB.Board:
+          return 'board';
+        default:
+          return 'document';
+      }
+    })();
+
+    navigate(`/page/${pageTypeRoute}/${page.id}`);
+  };
+
+  const onAddNewDocumentPage = async () => {
+    closePopup();
+    if (!workspace?.id) return;
+    const workspaceService = new WorkspaceBackendService(workspace.id);
+    const newViewResult = await workspaceService.createView({
+      name: 'New Document 1',
+      layoutType: ViewLayoutPB.Document,
+      parentViewId: page.id,
+    });
+    if (newViewResult.ok) {
+      try {
+        const newView = newViewResult.val;
+        const c = new DocumentController(newView.id);
+        await c.create();
+        await c.dispose();
+        appDispatch(
+          pagesActions.addPage({
+            parentPageId: page.id,
+            pageType: ViewLayoutPB.Document,
+            title: newView.name,
+            id: newView.id,
+            showPagesInside: false,
+          })
+        );
+        if (!page.showPagesInside) {
+          appDispatch(pagesActions.toggleShowPages({ id: page.id }));
+        }
+
+        navigate(`/page/document/${newView.id}`);
+      } catch (e) {
+        console.error(e);
+      }
+    }
+  };
+
+  const onAddNewBoardPage = async () => {
+    closePopup();
+    if (!workspace?.id) return;
+    const workspaceService = new WorkspaceBackendService(workspace.id);
+    const newViewResult = await workspaceService.createView({
+      name: 'New Board 1',
+      layoutType: ViewLayoutPB.Board,
+      parentViewId: page.id,
+    });
+
+    if (newViewResult.ok) {
+      const newView = newViewResult.val;
+      if (!page.showPagesInside) {
+        appDispatch(pagesActions.toggleShowPages({ id: page.id }));
+      }
+
+      appDispatch(
+        pagesActions.addPage({
+          parentPageId: page.id,
+          pageType: ViewLayoutPB.Board,
+          title: newView.name,
+          id: newView.id,
+          showPagesInside: false,
+        })
+      );
+
+      navigate(`/page/board/${newView.id}`);
+    }
+  };
+
+  const onAddNewGridPage = async () => {
```

Comment:
Maybe `onAddNewGridPage` and `onAddNewBoardPage` and `onAddNewDocumentPage` should be a `onAddPage`. The arguments are `type` `backUrl` etc.

Reply:
agreed